drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,634 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "rsi_rom_udma_wrapper.h"
+#include "rsi_udma.h"
+#include "sl_si91x_dma.h"
+#include "sl_status.h"
+
+#include <errno.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+
+#define DT_DRV_COMPAT          silabs_siwx917_dma
+#define DMA_PRIORITY_LOW       0
+#define DMA_PRIORITY_HIGH      1
+#define DMA_MAX_TRANSFER_COUNT 1024
+#define CHANNEL_12             12
+#define CHANNEL_32             32
+#define UDMA0_INSTANCE         0
+#define ULP_UDMA_INSTANCE      1
+#define PRIORITY_HIGH          1
+#define PRIORITY_LOW           0
+#define VALID_BURST_LENGTH     0
+#define UDMA_ADDR_INC_NONE     0X03
+
+LOG_MODULE_REGISTER(si91x_dma, CONFIG_DMA_LOG_LEVEL);
+
+static RSI_UDMA_HANDLE_T siwx917_udma_handle[2]; // Structure to store udma handles for UDMA0 and
```

Comment:
It can go, but during instantiation, dma_siwx917_data is created for each channel of the DMA. This may take up ax extra memory if I add this in dma_siwx917_data. But I will try to explore other means.

Reply:
Added this handle in dma_siwx917_data struct (dma_rom_buff[])